src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/DefaultDotNetProjectHost.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.Editor.Razor;
+using MonoDevelop.Projects;
+
+namespace Microsoft.VisualStudio.Mac.LanguageServices.Razor.ProjectSystem
+{
+    internal class DefaultDotNetProjectHost : DotNetProjectHost
+    {
+        private const string ExplicitRazorConfigurationCapability = "DotNetCoreRazorConfiguration";
+
+        private readonly DotNetProject _project;
+        private readonly ForegroundDispatcher _foregroundDispatcher;
+        private readonly VisualStudioMacWorkspaceAccessor _workspaceAccessor;
```

Comment:
Is there a good reason why this isn't `VisualStudioWorkspaceAccessor`?

Reply:
So is `DotNetProjectHost` a mono type?

Reply:
`DotNetProjectHost` is our own type that's VS4Mac specific. We can't use the normal VSWA because it doesn't allow us to take in a MonoDevelop solution and then retrieve the workspace for that solution.